### DISCUSSION TITLE
Desk stopped raising / lowering, even with new power supply -- T POWER printer power supply works, but only in 1 cm increments

### DISCUSSION THREAD

**Discussion by TheChuckster** [gcormier/megadesk#50] [General]
Need troubleshooting help -- this Megadesk has been working great for about a year, but now the desk no longer raises or lowers:

- Tried using connector-compatible 32V 2A printer power supply from T Power -- desk moves 1 cm then relay clicks, then requires power cycle again to move again: https://www.amazon.com/gp/product/B019HVZ4EE/

- Replaced with new ROL Ergo OEM power supply from IKEA, desk does not move in original mode, desk also doesn't move, but relay DOES making clicking sound in VARIANT mode

- Buttons seem to work

- Cleared EEPROM

- Obviously I have tried power-cycling many times

- I do not have the ORIGINAL BEKANT controller (I believe I discarded it) to recalibrate/reset the motors

Any ideas? I suspect my next step should be to try with the original controller (by purchasing a new frame again...) -- if that doesn't work, then I need to replace the motors / frame...

**Comment by gcormier** [comment]
This is a tough one as I do suspect calibration would help.

Can you try to lower the desk all the way to the bottom position as much as possible? (Realizing this will be rather painful and require many power off/on cycles and 1cm moves), and see if it behaves any better?

I think the next step would either be obtaining a replacement controller, or getting the calibration routine in megadesk. The latter is still on my todo list and will take some time, and then would also require you to flash the new firmware.

What size desk do you have, large or small?

**Comment by TheChuckster** [comment]
Do you have a spare original controller I can purchase? Would IKEA have one?

Do you also accept donations? -- really interested in seeing the calibration routine working ;-)

**Comment by gcormier** [comment]
At this point in time I think the code/flash is nearly maxed with serial functions enabled, but it's still on my to-do list. Unfortunately as it's rare to recalibrate and the stock ones does this, it's not super high on the priority list, but I will try to rig something up to capture the packets.

I know I read somewhere else that a user had success talking to IKEA support on the phone and getting them to send them another controller when theirs broke - have you tried calling them?

**Comment by TheChuckster** [comment]
I'll see what I can do. They are not responsive to phone calls. Will keep you posted once I manage to get my hands on one again and try recalibrating.

**Comment by foosel** [comment]
FWIW, years ago I wrote them an email (or maybe a support ticket? Sorry, memory is hazy, this was early 2016) if I could get a second controller due to issues my first one and I got one sent pretty quickly.

**Reply by gcormier** [reply]
@foosel  Any chance you still have the old controller, even if broken?

**Reply by foosel** [reply]
I disassembled it to figure out how it ticks, but somewhere I should still have the parts if I'm not mistaken. Why, want me to send it your way?

**Reply by gcormier** [reply]
Sure, I'm mostly interested in the grey housing/enclosure and buttons. Happy to cover shipping costs - shoot me an email and we can chat more?

**Comment by orther** [comment]
I had the same issue with the desk only going up in ~1cm increments until I read online if I unplug the desk for a day or so it will start to worm again. I unplugged it for two days and when I plugged it back in it started working like usual. 

The problem is if you leave it plugged in to power the desk starts doing to 1cm steps again. To solve my issue I ordered the following power foot pedal switch to prevent electricity from going to the desk until I step on the pedal.

https://www.amazon.com/dp/B08BZX4XGY/ref=cm_sw_r_oth_apip_J6hbX3L434T2V

Not sure if this is helpful but figured I'd share in case it was.

**Reply by TheChuckster** [reply]
Unfortunately, this wasn't helpful for my case. Would be interested in seeing how the new calibration update works for you though!

**Comment by gcormier** [comment]
Going to move this over to discussions. I do wonder if based on what orther said if it is a power supply issue - otherwise it could be something where the motors are refusing to move?

**Reply by TheChuckster** [reply]
I got my hands on a stock control pod. It works! So it's not a power supply issue. In fact, I am using the T Power power supply with the stock control pod, and it's working.

**Reply by gcormier** [reply]
@TheChuckster Can you run a re-calibration procedure with the stock controller?

Good news is the latest development builds have the calibration routine in them, so the original controller is no longer required in the future.

**Reply by TheChuckster** [reply]
Yes. Running the calibration on the stock control pod fixed my megadesk. I suspect I need to reflash my megadesk so that it natively has the calibration routine?

**Reply by gcormier** [reply]
Yes, flashing will be required. The new firmware is still under development, a few finishing touches I imagine. If you only want to flash once in a few weeks would probably be a good time.
